lib/reek/configuration/excluded_paths.rb

```diff
@@ -10,10 +11,13 @@ module Configuration
     module ExcludedPaths
       include ConfigurationValidator
 
-      # @quality :reek:NestedIterators { max_allowed_nesting: 2 }
+      # @param paths [String]
+      # @return [undefined]
       def add(paths)
-        paths.each do |path|
-          with_valid_directory(path) { |directory| self << directory }
+        paths.map { |path| Pathname(path) }.each do |path|
+          raise Errors::ConfigFileError, "Exclude path #{path} does not exist" unless path.exist?
```

Comment:
I'm not sure we want to do that since often people may add, e.g., editor artifacts to such exclusion lists, and they may or may not exist.

Reply:
I think it's advantageous for the user that we warn him with a bang instead of silently ignoring it.
Reek is all about code quality, why not enforce "configuration quality" as well?
I can't think of a good use case when a user would not want to be alerted of a useless configuration setting - do you have an example?

Reply:
I have, for example, a `tmp/` directory in my Reek checkout. It is git-ignored so it won't exist on a fresh checkout. It contains several sample Ruby files which I parked there to work on Reek issues. I can imagine adding `tmp/` as an ignored directory in `.reek.yml`, but wouldn't want Reek to fail on a fresh checked-out repository.

In general, it's mainly an issue when different people working on the same project have different local artifacts lying around in their repositories that they want to ignore but may not necessarily exist.

Another example is the `vendor/` directory. Notice that it is ignored in our `.rubocop.yml` config because it exists in CI. People might similarly run just `reek` in CI, but not want it to scan `vendor/`.

Reply:
But the downside is that you might have some configuration in your config file that you think is doing sth when in fact is doing nothing because you made a typo in the path you listed. We could fix this by adding a "strict" mode or similar in the future.

I do see your point though and will remobmve those checks.